dashboard/app/controllers/registrations_controller.rb

```diff
@@ -5,10 +5,22 @@ class RegistrationsController < Devise::RegistrationsController
     :migrate_to_multi_auth, :demigrate_from_multi_auth
   ]
   skip_before_action :verify_authenticity_token, only: [:set_age]
+  skip_before_action :clear_sign_up_session_vars, only: [:new, :create]
 
   def new
     session[:user_return_to] ||= params[:user_return_to]
     @already_hoc_registered = params[:already_hoc_registered]
+
+    unless session[:sign_up_tracking_expiration]&.future?
+      session[:sign_up_uid] = SecureRandom.uuid.to_s
+      session[:sign_up_tracking_expiration] = 5.minutes.from_now
+      FirehoseClient.instance.put_record(
```

Comment:
Tracking question: We're only putting the 'load-sign-up-page' event if we didn't already start tracking.  Are we intentionally squelching the case where a user hits the signup page then reloads it within the 5-minute window?  I'm wondering if this put_record should be outside the if-statement, or in a separate if-statement.